### PR TITLE
[ECO-2188] Revise indexer NAT gateway(s), rules, autoscaling

### PR DESCRIPTION
**Pull request by alnoki**
# Description

## `cfn-lint`

The following `cfn-lint` issues have been addressed, enabling template rules:

1. https://github.com/aws-cloudformation/cfn-lint/issues/3630
1. https://github.com/aws-cloudformation/cfn-lint/issues/3710

Hence:

1. Bump `cfn-lint` version in `pre-commit` config.
1. Convert cascaded `MaybeDeploy*` conditions to `Deploy*` rules with asserts.

## NAT gateways

1. Update NAT gateway architecture to have redundant NAT gateways per AWS best
   practices.
1. Create new private route tables and subnet associations for one NAT gateway
   per subnet.
1. Abstract associated resource creation using `Fn::ForEach` operator.

## Autoscaling

1. Abstract autoscaling targets and policies for broker and PostgREST.
1. Use target autoscaling for scaling out, and custom step scaling for scaling
   in, effectively silencing an erroneous alarm for low CPU usage on an idle
   single container.

## Miscellaneous

1. Update README line breaking and stack name setting commands.
1. Update README to reflect new simplified `Deploy*` parameters.
1. Document autoscaling and NAT gateway architectures.
1. Remove extraneous `!Ref` and/or `!Sub` commands where not needed.
1. Bump processor and broker binaries per recent code changes.
1. Remove general cache against `/*` due to wildcard issues.

# Testing

REST and WS public endpoints were verified before the binaries were updated in 74c71ee, which required a redeployment of the stack due to breaking migrations reversions. Once backfilled, the endpoints can be tested again.

# Checklist

- [x] Did you update relevant documentation?
- [x] Did you check all checkboxes from the linked Linear task?

### CODE REVIEW

File: src/cloud-formation/deploy-dev.yaml [hunk 1]
```diff
@@ -1,26 +1,26 @@
...
-  MaybeDeployNlbVpcLink: 'true'
-  MaybeDeployPostgrest: 'true'
-  MaybeDeployProcessor: 'true'
-  MaybeDeployRestApi: 'true'
-  MaybeDeployRestApiDnsRecord: 'true'
-  MaybeDeployRouteTables: 'true'
-  MaybeDeployStack: 'true'
-  MaybeDeployVpc: 'true'
-  MaybeDeployWaf: 'true'
+  Environment: 'ECO-2188'
```
Comment (alnoki) [inline src/cloud-formation/deploy-dev.yaml:23]: - [ ] Revert to `dev` before merge

File: src/cloud-formation/deploy-dev.yaml [hunk 2]
```diff
@@ -1,28 +1,28 @@
...
   Network: 'testnet'
-  ProcessorImageVersion: '0.5.0'
+  ProcessorImageVersion: '0.5.9'
```
Comment (xbtmatt) [inline src/cloud-formation/deploy-dev.yaml:25]:
```suggestion
  ProcessorImageVersion: '0.6.0'
```

You can use `0.6.0`, it's technically more up to date than `0.5.9` now!

**Review by xbtmatt — APPROVED** [review]
Just the processor version change, feel free to resolve the comment once updated

File: src/cloud-formation/deploy-dev.yaml (hunk 1 above)
Comment (alnoki) [reply src/cloud-formation/deploy-dev.yaml]: Addressed in d97df24cf84485b6268ff949989524114cd786d0

**Pull request merged**